src/cast_core/src/CrudActor.cpp

```diff
@@ -1132,6 +1135,87 @@ std::string getDbName(const PhaseContext& phaseContext) {
     return phaseDb ? *phaseDb : *actorDb;
 }
 
+enum Units { NANOSECOND, MICROSECOND, MILLISECOND, SECOND, MINUTE, HOUR };
+// represent a random delay
+class Delay {
+public:
+    Delay(const Node& node, GeneratorArgs args)
+        : number_generator{doubleGenerator(node["^TimeSpec"]["value"], args)} {
+        auto unit_string = node["^TimeSpec"]["units"].maybe<std::string>().value_or("seconds");
+
+        // Use string::find here so plurals get parsed correctly.
+        if (unit_string.find("nanosecond") == 0) {
```

Comment:
I suspect the real answer here is to make a class hierarchy, with a class per type. Or better yet a template, and add a factor function to produce them. I'm not sure I care enough to do that.

Reply:
> Or better yet a template, and add a factor function to produce them. I'm not sure I care enough to do that.

Ya this is entirely in a `.cpp` so 🤷🏽, but this impl is straightforward enough and not really evaluated in a loop so clarity is probably better than raw speed and cleverness.